src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowExtended.groovy

```diff
@@ -92,4 +106,78 @@ class HaFlowExtended {
                                boolean useTraffgenPorts = true, List<Integer> busyEndpoints = []) {
         return new HaFlowBuilder(swT, northboundV2, topologyDefinition, useTraffgenPorts, busyEndpoints)
     }
+
+    HaFlowUpdatePayload convertToUpdateRequest() {
+
+        String diverseLink = diverseWithFlows.isEmpty() ?
+                (diverseWithYFlows.isEmpty() ? (diverseWithHaFlows.isEmpty() ? null : diverseWithHaFlows[0])
+                        : diverseWithYFlows[0])
+                : diverseWithFlows[0]
+
+        def haFlowCopy = this.jacksonCopy()
```

Comment:
Let's use groovy's native mechanism (@Cloneable) for cloning the objects.

Reply:
Done.